### PR TITLE
ethtool: Add ethtool PAUSE support

### PR DESCRIPTION
**Pull request by cathay4t**
Example on `npc sim0` output against netdevsim nic:

```yaml
  ethtool:
    pause:
      rx: true
      tx: true
      auto_negotiate: false
```

Python binding also updated.

 Test case included but disabled for CI as CI system has no netdevsim kernel module

### CODE REVIEW

**Comment by cathay4t** [comment]
@ffmancera PR is ready for review.

**Comment by cathay4t** [comment]
The `netlink-generic` and `netlink-ethtool` crate will be removed once https://github.com/little-dude/netlink they merged them.

File: src/lib/ifaces/mac_vtap.rs [hunk 1]
```diff
@@ -5,8 +5,8 @@ use crate::NisporError;
 use netlink_packet_route::rtnl::link::nlas;
 use serde_derive::{Deserialize, Serialize};
 
-#[serde(rename_all = "lowercase")]
 #[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+#[serde(rename_all = "lowercase")]
```
Comment (ffmancera) [inline src/lib/ifaces/mac_vtap.rs]: I think this change is unrelated, is because of rustfmt? Does it matter now?

File: src/lib/ifaces/mac_vlan.rs [hunk 2]
```diff
@@ -16,8 +16,8 @@ const MACVLAN_MODE_BRIDGE: u32 = 4;
 const MACVLAN_MODE_PASSTHRU: u32 = 8;
 const MACVLAN_MODE_SOURCE: u32 = 16;
 
-#[serde(rename_all = "lowercase")]
 #[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+#[serde(rename_all = "lowercase")]
```
Comment (ffmancera) [inline src/lib/ifaces/mac_vlan.rs]: Same here..

File: src/python/nispor/ethtool.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
+# Copyright 2020 Red Hat
```
Comment (ffmancera) [inline src/python/nispor/ethtool.py]: Just a nit, 2021 :-)

**Review by ffmancera — CHANGES_REQUESTED** [review]
The file `src/netlink-ethtool/pause/message.rs` is something missing here or is this a typo? Please, adjust it.

I am also curious, rust files does not require a copyright statement? Thanks!

File: src/lib/ifaces/mac_vtap.rs (hunk 1 above)
Comment (cathay4t) [reply src/lib/ifaces/mac_vtap.rs]: The cargo clippy warn me that `serde` is used before derive introduced it.

Comment (cathay4t) [reply src/lib/ifaces/mac_vtap.rs]: Moved to https://github.com/nispor/nispor/pull/119

**Review by ffmancera — APPROVED** [review]
LGTM, thanks!
